x-pack/test/api_integration/apis/ml/results/get_category_definition.ts

```diff
@@ -0,0 +1,184 @@
+    examples: [
+      '130.246.123.197 - - [2018-07-22T03:26:21.326Z] "GET /beats/metricbeat HTTP/1.1" 200 6850 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '130.246.123.197 - - [2018-07-22T03:26:21.326Z] "GET /beats/metricbeat_1 HTTP/1.1" 200 6850 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '223.87.60.27 - - [2018-07-22T00:39:02.912Z] "GET /elasticsearch/elasticsearch-6.3.2.deb HTTP/1.1" 200 6219 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+      '223.87.60.27 - - [2018-07-22T00:39:02.912Z] "GET /elasticsearch/elasticsearch-6.3.2.deb_1 HTTP/1.1" 200 6219 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:6.0a1) Gecko/20110421 Firefox/6.0a1"',
+    ],
+  };
+
+  async function getCategoryDefinition(
+    jobId: string,
+    categoryId: string,
+    user: USER,
+    expectedStatusCode: number,
+    space?: string
+  ) {
+    const { body } = await supertest
+      .post(`${space ? `/s/${space}` : ''}/api/ml/results/category_definition`)
+      .auth(user, ml.securityCommon.getPasswordForUser(user))
+      .set(COMMON_REQUEST_HEADERS)
+      .send({ jobId, categoryId })
+      .expect(expectedStatusCode);
+
+    return body;
+  }
+
+  describe('get category_definition', function () {
+    before(async () => {
+      await esArchiver.loadIfNeeded('x-pack/test/functional/es_archives/ml/module_sample_logs');
+      await ml.testResources.setKibanaTimeZoneToUTC();
+      await spacesService.create({ id: idSpace1, name: 'space_one', disabledFeatures: [] });
+      await spacesService.create({ id: idSpace2, name: 'space_two', disabledFeatures: [] });
+
+      await ml.api.createAndRunAnomalyDetectionLookbackJob(
+        // @ts-expect-error not full interface
+        testJobConfig,
+        testDatafeedConfig,
+        idSpace1
+      );
+    });
+
+    after(async () => {
+      await spacesService.delete(idSpace1);
+      await spacesService.delete(idSpace2);
+      await ml.testResources.deleteIndexPatternByTitle('ft_module_sample_logs');
+      await ml.api.cleanMlIndices();
+    });
+
+    it('should produce the correct category for the job', async () => {
+      const resp = await getCategoryDefinition(
+        jobIdSpace1,
+        expectedCategoryDefinition.categoryId,
+        USER.ML_POWERUSER,
+        200,
+        idSpace1
+      );
+
+      expect(resp.categoryId).to.eql(
+        expectedCategoryDefinition.categoryId,
+        `categoryId should be ${expectedCategoryDefinition.categoryId} (got ${resp.categoryId})`
+      );
+      expect(resp.examples.length).to.eql(
+        expectedCategoryDefinition.examples.length,
+        `examples list length should be ${expectedCategoryDefinition.examples.length} (got ${resp.examples.length})`
+      );
+      expect(resp.terms.length).to.be.greaterThan(
+        0,
+        `terms string length should be greater than 0 (got ${resp.terms.length})`
+      );
```

Comment:
With the new assertions, we're not covering the `regex` part anymore. We could add a check that it's not empty or something.

Reply:
Updated in https://github.com/elastic/kibana/pull/124710/commits/7634363c0ce8420f1ce649f17012731ecce30cae